src/net.h

```diff
@@ -546,8 +546,8 @@ class CNode
     std::vector<CAddress> vAddrToSend;
     std::unique_ptr<CRollingBloomFilter> m_addr_known{nullptr};
     bool fGetAddr{false};
-    std::chrono::microseconds m_next_addr_send GUARDED_BY(cs_sendProcessing){0};
-    std::chrono::microseconds m_next_local_addr_send GUARDED_BY(cs_sendProcessing){0};
+    std::chrono::microseconds m_next_addr_send{0};
+    std::chrono::microseconds m_next_local_addr_send{0};
```

Comment:
These members should never have been guarded by `cs_sendProcessing`, which really only exists at this point to make sure that multiple threads don't enter `SendMessages()` concurrently.

The annotation was over-zealously added in #13123. The author of that PR agrees that these annotations are unnecessary: https://github.com/bitcoin/bitcoin/pull/13123#issuecomment-647505130.

These variables are only ever read/written by the message handler thread inside `SendMessages()`, so it's safe for them to be unguarded. I can make them atomic if you think that would make this more obviously safe.

Reply:
"These variables are only ever read/written ... inside SendMessages" -- is **exactly** what GUARDED_BY(cs_sendProcessing) documents. The point of having the guard is to catch errors when we change the code in future and violate the assumptions we were relying on -- if the code compiles with the GUARDED_BY statements, it will always also compile correctly without them -- the trouble comes when people make changes later.

I don't know what @practicalswift thinks the annotation was a mistake; it's not. (It's a mistake the other addr-related member variables aren't guarded though)

Having things be atomic instead of guarded is only a good idea if they're actually accessed by multiple threads, and either they don't need to be synchronized with any other data, or the timing is too important for proper locks.

Reply:
> It's a mistake the other addr-related member variables aren't guarded though

Ok. I've moved all of the addr fields to be guarded by their own `m_addr_mutex` mutex. This seems much better than having data/logic that should be in net_processing being locked by a mutex in net.

Reply:
That proved more difficult than expected. I've reverted that change and have moved just  `m_next_addr_send` and `m_next_local_addr_send` to be guarded by the new `m_addr_mutex`. I'll move the rest of the addr fields under that mutex when they move into the `Peer` struct.

Reply:
I don't follow why you're changing those guards at all in this PR; you don't need to. See https://github.com/ajtowns/bitcoin/tree/202003-pr21236

I don't think it will make sense to lock `m_addr_known` and `vAddrToSend` with per-peer locks: when receiving an addr message we'll obtain the lock for the peer that sends the message, then attempt to obtain the lock for a peer we want to forward the address on to -- but there's no particular reason to expect those peers to always be in a particular order, so holding the sending peer's lock would create ordering violations, and releasing and reacquiring would be a bit ridiculous given that this is always happening in a single thread.

Reply:
Thanks! Totally agree that taking and releasing multiple locks would be ridiculous. I've taken your branch.

Reply:
This is still causing confusion for reviewers, so I've moved the fields to have their own lock that is only used in net_processing.